aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -94,6 +94,31 @@ void div_kernel(TensorIterator& iter) {
   }
 }
 
+void remainder_kernel(TensorIterator& iter) {
+  if (isIntegralType(iter.dtype(), /*includeBool*/ false)) {
+    AT_DISPATCH_INTEGRAL_TYPES(iter.dtype(), "remainder_cpu", [&]() {
+      cpu_kernel(iter, [](scalar_t a, scalar_t b) -> scalar_t {
+        TORCH_CHECK(b != 0, "ZeroDivisionError");
+        scalar_t r = a % b;
+        if ((a != 0) && (a < 0) != (b < 0)) {
+          r += b;
+        }
+        return r;
+      });
+    });
+  } else {
+    AT_DISPATCH_FLOATING_TYPES_AND_HALF(iter.dtype(), "remainder_cpu", [&]() {
+      cpu_kernel_vec(iter,
+        [=](scalar_t a, scalar_t b) __ubsan_ignore_float_divide_by_zero__ -> scalar_t {
+          return a - b * at::native::floor_impl(a / b);
```

Comment:
the old implementation had a check against `b` being 0 and an explicit set to NaN in that case.  Why the change here?  Is this guaranteed to be the same?  What happens in the vectorized case? (does __ubsan_ignore_float_divide_by_zero__ apply there?)

Reply:
I didn't do an explicit check because the `div` operator doesn't explicitly check either. But I don't know if this is guaranteed to be the same, so I'll add the check back in.

I'm not sure about the vectorized case. I don't know how to make pytorch tensors use vectorized types.

I don't know what `__ubsan_ignore_float_divide_by_zero__` does, I just copied it from `div`.

Reply:
It uses vectorization if the array is big enough -- I just added some print statements and e.g.:
`torch.randn(2,3).remainder(0)` calls the scalar version and `torch.randn(200, 300).remainder(0)` calls the Vec256 version.

Reply:
ok, I looked into this a bit -- I'm not actually sure if the vectorized divide by zero is undefined or not -- we could test it and see -- just use the example I gave above.

Reply:
also in this case, we can avoid the undefined behavior by checking if b is 0, like TH and THC do.  That might even be faster?   Can you rerun the benchmarks and check?

Reply:
Ah ok, using large tensors worked for me too. I've added explicit float divide by zero checks.